src/transformers/models/llava_next/modeling_llava_next.py

```diff
@@ -511,52 +773,30 @@ def forward(
-                    if image_feature.shape[0] > 1:
-                        base_image_feature = image_feature[0]
-                        image_feature = image_feature[1:]
-
-                        if height * width != base_image_feature.shape[0]:
-                            raise ValueError("The number of patches is not consistent with the image size.")
-                        num_patch_height, num_patch_width = get_anyres_image_grid_shape(
-                            image_sizes[image_idx],
-                            self.config.image_grid_pinpoints,
-                            self.config.vision_config.image_size,
-                        )
-                        image_feature = image_feature.view(num_patch_height, num_patch_width, height, width, -1)
-                        image_feature = image_feature.permute(4, 0, 2, 1, 3).contiguous()
-                        image_feature = image_feature.flatten(1, 2).flatten(2, 3)
-                        image_feature = unpad_image(image_feature, image_sizes[image_idx])
-                        image_feature = torch.cat(
-                            (
-                                image_feature,
-                                self.image_newline[:, None, None].expand(*image_feature.shape[:-1], 1),
-                            ),
-                            dim=-1,
-                        )
-                        image_feature = image_feature.flatten(1, 2).transpose(0, 1)
-                        image_feature = torch.cat((base_image_feature, image_feature), dim=0)
-                    else:
-                        image_feature = image_feature[0]
-                        image_feature = torch.cat((image_feature, self.image_newline[None]), dim=0)
-                    new_image_features.append(image_feature)
-                image_features = torch.stack(new_image_features, dim=0)
-
-                inputs_embeds, attention_mask, labels, position_ids = self._merge_input_ids_with_image_features(
-                    image_features, inputs_embeds, input_ids, attention_mask, labels
+                image_features, feature_lens = self.pack_image_features(
+                    image_features,
+                    image_sizes,
+                    image_newline=self.image_newline,
                 )
-                if labels is None:
-                    labels = torch.full_like(attention_mask, self.config.ignore_index).to(torch.long)
+
+                inputs_embeds, attention_mask, position_ids, labels = self._merge_input_ids_with_image_features(
+                    image_features,
+                    feature_lens,
+                    inputs_embeds,
+                    input_ids,
+                    attention_mask,
+                    position_ids,
+                    labels=labels,
+                )
+
+            # pixel_values is not None but is empty ---> text only cases
+            elif pixel_values is not None and input_ids.shape[1] != 1 and pixel_values.size(0) == 0:
```

Comment:
This is a bit odd - I wouldn't expect us to have empty tensors being passed around. Is this coming from the processor?

Reply:
@nxphi47 to be addressed?

Reply:
@zucchini-nlp Under many distributed training paradigm, if the sample is a text-only cases, pixel_values cannot be None but an empty tensor. This is to support training with text-only cases mixed with vision cases

Reply:
will leave for @nxphi47 , if you have capacity to address it

Reply:
@zucchini-nlp @amyeroberts In some parallelism paradigm, if a batch of 2 samples, one has an image and the other one has no image. Then, if we split the batch into 2 minibatch and send to 2 gpus with `dist.broadcast`, the `pixel_values` of the no-image batch must be a tensor and cannot be None. In distributed broadcasting, can only broadcast gpu tensor and not None.

Still, you may revert this change and assume that any trainer will set pixel_values to None before sending to `model.forward`.

If you go this path, @zucchini-nlp can help revert it.

Reply:
FYI, The broadcasting things will happen if we set `dataset(streaming=True)`

Reply:
Thanks for clarifying @nxphi47. OK, in this case, let's keep the condition